src/Compilers/CSharp/Portable/Symbols/Source/FieldSymbolWithAttributesAndModifiers.cs

```diff
@@ -336,6 +336,15 @@ internal sealed override bool HasSpecialName
             }
         }
 
+        internal sealed override bool NonNullTypes
+        {
+            get
+            {
+                // PROTOTYPE(NullableReferenceTypes): temporary solution to avoid cycle
+                return SyntaxBasedNonNullTypes(this.AttributeDeclarationSyntaxList) ?? base.NonNullTypes; 
```

Comment:
@AlekseyTs In case this helps, this is the approach we had previously used for syntax-based detection of NonNullTypes.
The specific `SyntaxBasedNonNullTypes` in this PR was very inefficient, but we could use that model and make the check cheaper (ie. avoid `.ToString()`).